vendor/engines/secure_rooms/app/services/secure_rooms/access_rules/account_selection_rule.rb

```diff
@@ -0,0 +1,19 @@
+module SecureRooms
+
+  module AccessRules
+
+    class AccountSelectionRule < BaseRule
+
+      def self.condition(_user, _card_reader, accounts, selected)
+        if selected.present?
+          Verdict.new(:grant)
+        elsif accounts.present? && selected.blank?
+          Verdict.new(:pending, "Must select Account")
+        end
```

Comment:
I'm also still leaning a bit towards doing the account fetching here, if only to save the `accounts` argument. You could always return the `accounts` as part of the `pending` verdict.